polygeist/test/polygeist-opt/sycl/convertToLLVMABI.mlir

```diff
@@ -0,0 +1,13 @@
+// RUN: polygeist-opt --convert-to-llvm-abi --split-input-file %s | FileCheck %s
+
+// CHECK: gpu.func @kernel([[A0:%.*]]: !llvm.ptr<i32, 1>, %arg1: !sycl_range_1_, %arg2: !sycl_range_1_, %arg3: !sycl_id_1_) 
+// CHECK-SAME: kernel attributes {llvm.cconv = #llvm.cconv<spir_kernelcc>, llvm.linkage = #llvm.linkage<weak_odr>, passthrough = ["norecurse", "nounwind", "convergent", "mustprogress"]} {
+// CHECK-NEXT:      [[P2M:%.*]] = "polygeist.pointer2memref"([[A0]]) : (!llvm.ptr<i32, 1>) -> memref<?xi32, 1>
+// CHECK-NEXT:      sycl.call([[P2M]]) {Function = @foo, MangledName = @foo} : (memref<?xi32, 1>) -> ()
+
+gpu.module @device_functions {
+gpu.func @kernel(%arg0: memref<?xi32, 1>, %arg1: !sycl.range<1>, %arg2: !sycl.range<1>, %arg3: !sycl.id<1>) kernel attributes {llvm.cconv = #llvm.cconv<spir_kernelcc>, llvm.linkage = #llvm.linkage<weak_odr>, passthrough = ["norecurse", "nounwind", "convergent", "mustprogress"]} {
```

Comment:
add a test for a gpu.func returning a value.

Reply:
Can you ensure we also have a test containing a declaration of a gpu.func (rather than a definition), we need to check that declarations are converted correctly. 

I am not sure whether in MLIR we can have both a declaration and a definition for the same function in the same module. If that is allowed add a test for that scenario as well pls.

Reply:
> add a test for a gpu.func returning a value.

Will do that when adding the handling of return type.

> a test containing a declaration of a gpu.func

Working on it.

Reply:
ok doing it in a separate PR is fine